sql/core/src/main/java/org/apache/spark/sql/sources/v2/SessionConfigSupport.java

```diff
@@ -25,7 +25,7 @@
  * session.
  */
 @InterfaceStability.Evolving
-public interface SessionConfigSupport {
+public interface SessionConfigSupport extends DataSourceV2 {
```

Comment:
Why does this need to extend DataSourceV2? Why add this in a commit that appears to be nothing more than a rename?

Reply:
+1 for @rdblue 's question.
@cloud-fan . Could you add more explanation about this change at PR description?

Reply:
This is something we left behind. For a mix-in interface, it should extend the interface it aimed to mix in, so that we can guarantee, for example, classes implement `SessionConfigSupport` must also implement `DataSourceV2`.

We already did it in some mix-in interfaces, e.g. the streaming read support, `SupportsScanUnsafeRow`, `SupportsScanColumnarBatch`, etc. In this PR I just wanna fix the remaining ones, to make them consistent.

If you feel strongly about it, I can create a JIRA ticket for it.

Reply:
It's best to keep commits clean and focused. I'd say create a new JIRA for it and do all of the mix-ins at once.

+1 when this is separated to its own PR. Thanks!

Reply:
Created https://issues.apache.org/jira/browse/SPARK-23262 .

I'd like to keep the changes in this PR, as it's just several lines. It's not ideal but we are rushing for the 2.3 release, I wanna speed this up a little, what do you think?

Reply:
Ping me on the new PR. I'm happy to review it (though it is non-binding).

Reply:
@cloud-fan is sleeping. Let me submit the PR for resolving your concern. Thanks!